cmd/run.go

```diff
@@ -2,106 +2,49 @@ package cmd
-	"github.com/odpf/meteor/internal/metrics"
-	"github.com/odpf/meteor/plugins"
+	"github.com/odpf/meteor/metrics"
 	"github.com/odpf/meteor/recipe"
 	"github.com/odpf/meteor/registry"
+	"github.com/odpf/salt/log"
 	"github.com/spf13/cobra"
 )
 
-// RunCmd creates a command object for the "run" action
-func RunCmd() *cobra.Command {
-	cmd := &cobra.Command{
+// RunCmd creates a command object for the "run" action.
+func RunCmd(lg log.Logger, mt *metrics.StatsdMonitor) *cobra.Command {
+	return &cobra.Command{
 		Use:     "run [COMMAND]",
-		Short:   "Run meteor for provided receipes",
 		Example: "meteor run recipe.yaml",
+		Short:   "Run meteor for provided recipes.",
 		Args:    cobra.ExactArgs(1),
-		Run: func(cmd *cobra.Command, args []string) {
-			fi, err := os.Stat(args[0])
-			if err != nil {
-				fmt.Println(err)
-				return
-			}
-			switch mode := fi.Mode(); {
-			case mode.IsDir():
-				rundir(args[0])
-			case mode.IsRegular():
-				run(args[0])
-			}
-		},
-	}
-
-	return cmd
-}
+		RunE: func(cmd *cobra.Command, args []string) error {
 
-// Run a single recipe
-func run(recipeFile string) {
-	c, err := config.LoadConfig()
-	if err != nil {
-		log.Fatal(err)
-	}
-	log := logger.New(c.LogLevel)
-	plugins.Log.Level = log.Level
-
-	reader := recipe.NewReader()
-	rcp, err := reader.Read(recipeFile)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	run := initRunner(c, log).Run(rcp)
-	if run.Error != nil {
-		log.Fatal(run.Error)
-	}
-	log.WithField("run", run).Info("Done!")
-}
+			runner := agent.NewAgent(registry.Extractors, registry.Processors, registry.Sinks, mt)
 
-// Run a directory of recipes
-func rundir(dirPath string) {
-	c, err := config.LoadConfig()
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	reader := recipe.NewReader()
-	recipeList, err := reader.ReadDir(dirPath)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	log := logger.New(c.LogLevel)
-	plugins.Log.Level = log.Level
+			reader := recipe.NewReader()
+			recipes, err := reader.Read(args[0])
+			if err != nil {
+				return err
+			}
 
-	runs := initRunner(c, log).RunMultiple(recipeList)
-	log.WithField("runs", runs).Info("Done!")
-}
+			count := len(recipes)
 
-func initRunner(config config.Config, logger plugins.Logger) (runner *agent.Agent) {
-	metricsMonitor := initMetricsMonitor(config)
-	runner = agent.NewAgent(
-		registry.Extractors,
-		registry.Processors,
-		registry.Sinks,
-		metricsMonitor,
-	)
-	return
-}
+			if count == 0 {
+				lg.Info(fmt.Sprintf("no recipe found for path: %s", args[0]))
+				return nil
+			}
 
-func initMetricsMonitor(c config.Config) *metrics.StatsdMonitor {
-	if !c.StatsdEnabled {
-		return nil
-	}
+			runs := runner.RunMultiple(recipes)
 
-	client, err := metrics.NewStatsdClient(c.StatsdHost)
-	if err != nil {
-		panic(err)
+			for _, run := range runs {
+				if run.Error != nil {
+					lg.Error(fmt.Sprintf("%s: %s", run.Recipe.Name, run.Error))
+					continue
+				}
+				lg.Info(fmt.Sprintf("%s: success", run.Recipe.Name))
```

Comment:
```
lg.Info("success", "name", run.Recipe.Name)
```
the first argument is a string explaining what the log is, rest of the args are optional and alternates at key/value pairs.
Args key should be string and value could be anything printable.